cluster/encryption.go

```diff
@@ -0,0 +1,293 @@
+package cluster
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+
+	normantypes "github.com/rancher/norman/types"
+	"sigs.k8s.io/yaml"
+
+	"github.com/rancher/rke/k8s"
+	"github.com/rancher/rke/log"
+	"github.com/rancher/rke/services"
+	"github.com/rancher/rke/templates"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	apiserverconfig "k8s.io/apiserver/pkg/apis/config/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	EncryptionProviderFilePath = "/etc/kubernetes/ssl/encryption.yaml"
```

Comment:
will this file be always static? the location, the name?

Reply:
Yes. This is the rke "managed" file location. And it's safer to control the file name.